Components/Resource.tsx

```diff
@@ -0,0 +1,63 @@
+
+import React, { FC } from 'react';
+import styles from '../styles/Home.module.css'
+import { ResourcesData } from '../pages/resources';
+
+interface EventProps {
+    resourcesData: ResourcesData[]
+    headingColumns: string[]
+}
+  
+const Event: FC<EventProps> = ({resourcesData}: EventProps, headingColumns) => {
+    const event = (resourcesData as Array<ResourcesData>).map((row, index) => {
+      let rowData: { key: string; val: string | number; }[] = [];
+
+      Object.entries(row).forEach((data, i) => {
+        rowData.push({
+        key: headingColumns[i],
+        val: data[1]
+        });
+      });
+
+      const displayDetails = (data) => {
+        alert(`${data}`);
+      }
+
+      return <div className={styles.event_container_1r} key={index}>
```

Comment:
Ok to answer the black red and orientation questions, what I would do is I wouldn't return the div with all of the row data immediately.
Instead, I would create a resources array and in the map arrow function, I would pass in (data, index) as you're already doing and check if (index + 1) %3 == 1. If it is, that means there are already 3 events on that row, and that it's time to reverse the colors and orientation.
Here's some pseudo-code for ya:
var color = black; //
var orientation = left;

    rowData......map(data, index) => {
        if(index % 3 == 1) {
        //change color and orientation to opposite of previous var values
        // i.e color = red;
        }
        else {
        //do nothing
        }

       if(color = red && orientation = right) { 
       resources[index] = <div className={styles.black_and_left_orientation_event} .... </div>
       }
       //or when its the different color and orientation,
       resources[index] =<div className={styles.red_and_right_orientation_event} .... </div>
    }
}

... and then I would return the resources array with map in your newly created "Resources" component/file

```
return (
<div>
{resources.map((r) => (
<div>{r}</div>
))}
</div>
);
```
I wouldn't bet on my array syntax, I think you can have it behave like an ArrayList in js, so using push(index) is prob the way to go, but this is how I would go about it.

for more info:

https://www.pluralsight.com/guides/how-to-use-the-map()-function-to-export-javascript-in-react